src/github.com/getlantern/chained/server.go

```diff
@@ -69,7 +69,7 @@ func (s *Server) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 		wg.Done()
 	}()
 	go func() {
-		go io.Copy(connIn, connOut)
+		io.Copy(connIn, connOut)
```

Comment:
The wg.Wait() was not actually doing nothing as the io.Copy was spawned into a goroutine. Now the wg.Wait will actually wait that both io.Copy are completed before returning control to the parent function.